src/rgw/rgw_sal.cc

```diff
@@ -125,21 +125,23 @@ int RGWRadosBucket::remove_bucket(bool delete_children, std::string prefix, std:
     }
   } while(is_truncated);
 
-  ret = abort_bucket_multiparts(store, store->ctx(), info, prefix, delimiter);
-  if (ret < 0) {
-    return ret;
+  if (!prefix.empty()) {
```

Comment:
@dang @cbodley @mattbenjamin : I'm trying to understand why we only clean up multiparts if prefix is empty.

I found the problem causing https://tracker.ceph.com/issues/52662 and was checking pacific and master if there was the same issue. That specific issue is resolved in master, but this conditional jumped out at me. Can anyone explain? When someone issues `radosgw-admin bucket rm --bucket=bkt1 --purge-objects --bypass-gc` this code gets run, and I don't think there's a prefix set and we should be cleaning up abandoned multipart uploads.

Reply:
@ivancich yeah this `!prefix.empty()` condition doesn't make sense to me either. in the comment linked above, @dang points out that `RGWDeleteBucket::execute()` has some swift-specific logic that may provide a prefix/delimiter when calling `remove_bucket()`. that logic was added long ago in https://github.com/ceph/ceph/pull/15630, but i don't see any explanation of that swift stuff in the pr or tracker issue

the openstack api docs for [container deletion](https://docs.openstack.org/api-ref/object-store/?expanded=delete-container-detail#delete-container) don't shed any light either..

> DELETE /v1/{account}/{container}
> Deletes an empty container. This operation fails unless the container is empty. An empty container has no objects.
> ...
> The container name cannot contain a slash (/) character because this character delimits the container and object name. For example, the path /v1/account/www/pages specifies the www container, not the www/pages container.

i say we remove that swift logic (and verify that nothing breaks in tempest tests), remove the prefix/delimiter arguments to both `remove_bucket()` and `abort_multiparts()`, and make this call to `abort_multiparts()` unconditional

Reply:
I don't remember why this was added, sorry, but I believe it must have been a result of failed tests.

Reply:
ok, it sounds like we need to reproduce that failure to understand it better. multipart is a s3 feature, so i wouldn't expect this branch to have any effect on swift tests

Reply:
I'm happy to take it on and see what happens a) when the condition is removed, and b) when prefix and delimiter are removed to see what happens.